src/app_service/service/contacts/service.nim

```diff
@@ -297,25 +297,47 @@ QtObject:
       else:
         contact.blocked = false
 
-      let response = status_contacts.addContact(contact.id, contact.name)
+      let response = status_contacts.sendContactRequest(contact.id, message)
       if(not response.error.isNil):
         let msg = response.error.message
-        error "error adding contact ", msg
+        error "error sending contact request", msg
         return
       self.saveContact(contact)
       self.events.emit(SIGNAL_CONTACT_ADDED, ContactArgs(contactId: contact.id))
     except Exception as e:
-      error "an error occurred while edding contact ", msg=e.msg
+      error "an error occurred while sending contact request", msg=e.msg
 
-  proc rejectContactRequest*(self: Service, publicKey: string) =
-    var contact = self.getContactById(publicKey)
-    contact.removed = true
+  proc acceptContactRequest*(self: Service, publicKey: string) =
+    try:
+      # NOTE: publicKey used for accepting last request
+      let response = status_contacts.acceptLatestContactRequestForContact(publicKey)
+      if(not response.error.isNil):
+        let msg = response.error.message
+        error "error accepting contact request", msg
+        return
+
+      # TODO: does we need this logic on both (sendContactRequest and acceptContactRequest) sides?
+      var contact = self.getContactById(publicKey)
+      if not contact.added:
+        contact.added = true
+      else:
+        contact.blocked = false
```

Comment:
If you accept a request from a contact and it is added, it gets blocked ?
What is the application use case ?

Reply:
I've saved it from older implementation, probably @saledjenic knows do we still need this logic. But it works different: if you add blocked contact - unblock it

Reply:
Ahh, I see. Sounds ok.